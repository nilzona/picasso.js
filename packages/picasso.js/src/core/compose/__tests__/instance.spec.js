import create from '../instance';

describe('instance', () => {
  let sandbox;
  let component;
  let data;
  let instance;

  beforeEach(() => {
    sandbox = sinon.createSandbox();
    component = sandbox.stub().returns({});
    data = sandbox.stub();
    instance = create({}, { registries: { component, data } });
  });

  afterEach(() => {
    sandbox.restore();
  });

  it('should create default instance', () => {
    // make a snapshot of default, update as we continue to add functionality
    expect(instance).toMatchSnapshot();
  });

  it('should add child', () => {
    const child = {};
    instance.addChild(child);
    expect(instance.getChildren().pop()).to.equal(child);
  });

  it('should get children', () => {
    const c1 = {};
    const c2 = {};
    const c3 = {};
    [c1, c2, c3].map(c => instance.addChild(c));
    expect(instance.getChildren()).to.have.length(3);
  });

});
