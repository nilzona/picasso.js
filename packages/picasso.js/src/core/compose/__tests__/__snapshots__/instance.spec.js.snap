// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'should create default instance' 1`] = `
Object {
  "addChild": [Function],
  "beforeDestroy": [Function],
  "beforeMount": [Function],
  "beforeRender": [Function],
  "beforeUnmount": [Function],
  "beforeUpdate": [Function],
  "componentDidMount": [Function],
  "componentDidUpdate": [Function],
  "componentWillMount": [Function],
  "componentWillReceiveProps": [Function],
  "componentWillUnmount": [Function],
  "componentWillUpdate": [Function],
  "created": [Function],
  "ctx": Object {
    "data": Array [],
    "formatter": undefined,
    "logger": undefined,
    "renderer": "svg",
    "style": undefined,
  },
  "destroyed": [Function],
  "getChildren": [Function],
  "getPreferredSize": [Function],
  "layoutComponents": [Function],
  "mounted": [Function],
  "render": [Function],
  "shouldComponentUpdate": [Function],
  "update": [Function],
  "updated": [Function],
  "visible": true,
}
`;
